client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -1019,48 +1019,48 @@ public String enumerateExtensions()  {
         return getResponse(String.class, clientResponse);
     }
 
-    public String unregisterExtension(final String extensionName) {
+    public APIResult unregisterExtension(final String extensionName) {
         ClientResponse clientResponse = new ResourceBuilder()
                 .path(ExtensionOperations.UNREGISTER.path, extensionName)
                 .call(ExtensionOperations.UNREGISTER);
-        return getResponse(String.class, clientResponse);
+        return getResponse(APIResult.class, clientResponse);
     }
 
-    public String getExtensionDetail(final String extensionName) {
-        return getResponse(String.class, getExtensionDetailResponse(extensionName));
+    public APIResult getExtensionDetail(final String extensionName) {
```

Comment:
Nothing to do with your patch, but, while you are at it, can you add this method to AbstractFalconClient too, please?